File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ha, hb = dict(), dict()
        for i in s:
            if i in ha.keys():
                ha[i] += 1
            else:
                ha[i] = 1
        for j in t:
            if j in hb.keys():
                hb[j] += 1
            else:
                hb[j] = 1
        if ha == hb:
            return True
        return False

File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        int n = s.length();
        if (n == 0) return false;
        int start = 0;
        int last = n - 1;
        while (start <= last) {
            char a = s.charAt(start);
            char b = s.charAt(last);
            if (!Character.isLetterOrDigit(a)) {
                start++;
            } else if (!Character.isLetterOrDigit(b)) {
                last--;
            } else {
                if (Character.toLowerCase(a) != Character.toLowerCase(b)) {
                    return false;
                }
                start++;
                last--;
            }
        }
        return true;
    }
}
